src/demo/ShowcaseGrid/ShowcaseGrid.tsx

```diff
@@ -0,0 +1,123 @@
+import React, {useMemo} from 'react';
+import {Text} from '../../components/Text';
+import {getPropsCombinations, PropCombination, PropSequences} from './getPropsCombinations';
+
+type Combinations<ComponentType extends React.ElementType> = PropCombination<
+    Partial<React.ComponentProps<ComponentType>>
+>;
+
+export function ShowcaseGrid<ComponentType extends React.ElementType>({
```

Comment:
This is an attempt to reduce and standardize the rendering of components with different combinations of props. It's not very optimal, just something like how one would do.